### ISSUE TITLE
Incorrect tab file name

### ISSUE THREAD

**Issue by marilmanen** [eteran/nedit-ng#224]
I have found two simple ways to get a wrong name for a tab. First includes usage of one of my macro commands. Here is the simplified macro.
```
    shell_command("rm " $file_name, "")
    focus_window($file_name)
```
The focus_window was my trial to avoid issues, but it does not help. Sequence to get a wrong name to a tab is.
1. open 3 files
2. run the macro for two of the files and do not click on the removed tab after macro execution
3. select the third file and press Ctrl-N (you should now get a pop-up warning of the first removed file)
4. press save on "Another program may have removed it" message window
Now the file name for the second removed file should be a duplicate of the third file and not the same as you see in the Windows menu. This sequence does not work in old NEdit as with it is not possible to switch out from a tab without getting the "Another program may have removed it" message window.

Same issue can be reproduced also without macro commands.
1. open 3 files
2. select file in the first tab
3. remove in a terminal window the two other files
4. activate nedit and press Ctrl+N.
5. Press save for the "Another program may have removed it" message window and the tab names will be wrong, but if you use the Cancel button the tab names remain correct.

**Comment by marilmanen** [comment]
Additional note: The New Untitled must become first tab. For example if Sort Tabs Alphabetically is set and the files are f1, f2 and f3, then Untitled will become first tab and the issue can be reproduced.

**Comment by eteran** [comment]
Very interesting, OK, I'll replicate locally and see if I can come up with a fix soon!

**Comment by eteran** [comment]
I'm not reproducing locally. I don't even get the prompt for the "Another program may have removed it" after step 4. I would if I selected the tabs for the deleted items, but you specifically said to select the first, and then delete the second and third.



https://user-images.githubusercontent.com/571477/103144864-9cb9ec80-46fe-11eb-9292-62a3d8ef9503.mp4

**Comment by eteran** [comment]
OK, I just reproduced! It is easily triggered with tab alphabetization enabled!


https://user-images.githubusercontent.com/571477/103144931-a42dc580-46ff-11eb-9798-f52997b780a4.mp4

**Comment by eteran** [comment]
@marilmanen OK, I think I have a fix! Just a simple reordering of operations to make sure we aren't renaming tabs based on index right before they are going to be moved around by alphabetization!

Can you give it a test?

https://github.com/eteran/nedit-ng/pull/225

**Comment by marilmanen** [comment]
I'm not able to reproduce the issue anymore, so I think the fix works as expected.

**Comment by eteran** [comment]
Nice!
